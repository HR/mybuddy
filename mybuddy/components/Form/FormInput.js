import React, { Component } from 'react'
import { Item, Label, Input, Text } from 'native-base'
import { StyleSheet } from 'react-native'

const capitalize = s => {
  if (typeof s !== 'string') return ''
  return s.charAt(0).toUpperCase() + s.slice(1)
}

export default class FormInput extends Component {
  _updateFormik = value => {
    const {
      field,
      form: { setFieldValue }
    } = this.props

    try {
      if (this.props.keyboardType === 'numeric') {
        const number = Number(value)
        if (number || !value) {
          setFieldValue(field.name, number || '')
        }
      } else {
        setFieldValue(field.name, value)
      }
    } catch (error) {}
  }

  render() {
    const {
      field, // { name, value, onChange, onBlur }
      form: { touched, errors, values }, // touched, errors, setXXXX, handleXXXX, dirty, isValid, status, etc.
      required,
      ...props
    } = this.props
    return (
      <>
        <Item
          floatingLabel
          style={styles.item}
          error={Boolean(touched[field.name] && errors[field.name])}
        >
          <Label>
            {`${props.label || capitalize(field.name)}${required ? '*' : ''}`}
          </Label>
          <Input
            value={props.value || String(values[field.name])}
            onChangeText={this._updateFormik}
            {...props}
          />
        </Item>
        {touched[field.name] && errors[field.name] && (
          <Text style={[styles.item, { color: 'red' }]}>
            {errors[field.name]}
          </Text>
        )}
      </>
    )
  }
}

const styles = StyleSheet.create({
  item: {
    alignSelf: 'center',
    width: '90%',
    marginTop: 10
  }
})
