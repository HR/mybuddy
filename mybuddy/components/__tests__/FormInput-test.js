import 'react-native'
import React from 'react'
import FormInput from '../Form/FormInput'
import renderer from 'react-test-renderer'

const props = (value, touched, error) => ({
  field: {
    name: 'test'
  },
  form: {
    values: {
      test: value
    },
    touched: {
      test: touched
    },
    errors: {
      test: error
    }
  }
});

describe('DateTimePicker snapshot', () => {
  jest.useFakeTimers()
  Date.now = jest.fn(() => 1503187200000)
  it('form input renders without errors', async () => {
    const tree = renderer
      .create(<FormInput {...props(null, false, false)} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

  describe('Form input renders with errors', () => {
    jest.useFakeTimers()
    Date.now = jest.fn(() => 1503187200000)
    it('form input renders with errors', async () => {
      const tree = renderer
        .create(<FormInput {...props(null, true, true)} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
