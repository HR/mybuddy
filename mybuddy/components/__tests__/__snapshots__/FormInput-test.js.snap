// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker snapshot form input renders without errors 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "transparent",
      "borderBottomWidth": 1,
      "borderColor": "#D9D5DC",
      "borderLeftWidth": 0,
      "borderRadius": undefined,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "borderWidth": undefined,
      "flexDirection": "row",
      "marginLeft": 2,
      "marginTop": 10,
      "opacity": 1,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "left": 0,
        "opacity": 1,
        "paddingBottom": undefined,
        "paddingTop": undefined,
        "position": "absolute",
        "right": 0,
        "top": 18,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#575757",
          "fontSize": 17,
          "paddingRight": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        float={true}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontSize": 15,
              "lineHeight": 30,
            },
          ]
        }
      >
        Test
      </Text>
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    editable={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholderTextColor="#575757"
    style={
      Object {
        "color": "#000",
        "flex": 1,
        "fontSize": 17,
        "height": 50,
        "paddingBottom": 7,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 3,
        "top": 8,
      }
    }
    underlineColorAndroid="rgba(0,0,0,0)"
    value="null"
  />
</View>
`;

exports[`Form input renders with errors form input renders with errors 1`] = `
Array [
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 1,
        "borderColor": "#ed2f2f",
        "borderLeftWidth": 0,
        "borderRadius": undefined,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "borderWidth": undefined,
        "flexDirection": "row",
        "marginLeft": 2,
        "marginTop": 10,
        "opacity": 1,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "left": 0,
          "opacity": 1,
          "paddingBottom": undefined,
          "paddingTop": undefined,
          "position": "absolute",
          "right": 0,
          "top": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#575757",
            "fontSize": 17,
            "paddingRight": 5,
            "paddingTop": 5,
          }
        }
      >
        <Text
          float={true}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontSize": 15,
                "lineHeight": 30,
              },
            ]
          }
        >
          Test
        </Text>
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#575757"
      style={
        Object {
          "color": "#000",
          "flex": 1,
          "fontSize": 17,
          "height": 50,
          "paddingBottom": 7,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 3,
          "top": 8,
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value="null"
    />
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "alignSelf": "center",
          "marginTop": 10,
          "width": "90%",
        },
        Object {
          "color": "red",
        },
      ]
    }
    uppercase={false}
  />,
]
`;
